### PR TITLE
Adding image input flags for new converter path

### PR DESCRIPTION
**Pull request by bhushan23**

### CODE REVIEW

File: tfcoreml/_tf_coreml_converter.py [hunk 1]
```diff
@@ -568,30 +603,56 @@ def convert(tf_model_path,
...
+      Supported iOS version options: '11.2', '12', '13'        
+      CoreML model produced by the converter will be compatible with the iOS version specified in this argument.
+      e.g. if target_ios = '12', the converter would only utilize CoreML features released till iOS12 (equivalently macOS 10.14, watchOS 5 etc).
+
+      iOS 11.2 (CoreML 0.8) does not support resize_bilinear, crop_resize layers 
+        - (Supported features: https://github.com/apple/coremltools/releases/tag/v0.8)
+      iOS 12 (CoreML 2.0)
+        - (Supported features: https://github.com/apple/coremltools/releases/tag/v2.0)
+      iSO 13 (CoreML 3.0)
+        - (Supported features: https://github.com/apple/coremltools/releases/tag/3.0-beta6)
```
Comment (1duo) [inline tfcoreml/_tf_coreml_converter.py]: Link to tag/3.0

Comment (bhushan23) [reply tfcoreml/_tf_coreml_converter.py]: done

File: tfcoreml/_tf_coreml_converter.py [hunk 2]
```diff
@@ -568,30 +603,56 @@ def convert(tf_model_path,
...
-  use_coreml_3: bool
-      Flag to switch to coremltools.converters.tensorflow which uses new Core ML 3 features.
+    
+  custom_shape_functions: dict(): {Text: func()}
+      Argument to provide user-defined functions to compute shape for given op.
+      A dictionary with keys corresponding to the type of TF Op and value as hadnled to user-defined function.
+      Function receives `layer specification` and `input shape` as a input.
+      output of the function must be output shape for give op. (generally List).
+      Custom shape function is required for adding custom layer in Core ML 3.
+      If use_coreml_3 is False, this functions are ignored
```
Comment (aseemw) [inline tfcoreml/_tf_coreml_converter.py]: `use_coreml_3` is not there anymore.

File: tfcoreml/_tf_coreml_converter.py [hunk 3]
```diff
@@ -568,30 +603,56 @@ def convert(tf_model_path,
...
-
-  use_coreml_3: bool
-      Flag to switch to coremltools.converters.tensorflow which uses new Core ML 3 features.
+    
+  custom_shape_functions: dict(): {Text: func()}
+      Argument to provide user-defined functions to compute shape for given op.
+      A dictionary with keys corresponding to the type of TF Op and value as hadnled to user-defined function.
+      Function receives `layer specification` and `input shape` as a input.
+      output of the function must be output shape for give op. (generally List).
+      Custom shape function is required for adding custom layer in Core ML 3.
```
Comment (aseemw) [inline tfcoreml/_tf_coreml_converter.py]: Custom shape function is required for adding custom layer when `target_ios` is greater than '12'. 
Couple of things:
- raise an error right away if custom_conversion_fucntion is provided but the shape function has not been given
- add a test that uses both these arguments and point to that test as an example in this description or in the readme in general

File: tests/test_pretrained_models.py [hunk 4]
```diff
@@ -354,6 +362,35 @@ def test_googlenet_v1_slim(self):
         output_tensor_name = 'InceptionV1/Logits/Predictions/Softmax:0',
         img_size = 224)
 
+  @unittest.skipIf(macos_version() < MIN_MACOS_VERSION_10_15,
+                     'macOS 10.15+ required. Skipping test.')
+  def test_googlenet_v1_slim_coreml_3(self):
```
Comment (aseemw) [inline tests/test_pretrained_models.py]: nice!

**Comment by aseemw** [comment]
make sure you run the test locally before merging this PR.

File: tfcoreml/_tf_coreml_converter.py (hunk 2 above)
Comment (bhushan23) [reply tfcoreml/_tf_coreml_converter.py]: done

File: tfcoreml/_tf_coreml_converter.py (hunk 3 above)
Comment (bhushan23) [reply tfcoreml/_tf_coreml_converter.py]: - `custom_conversion_function` does not require `custom_shape_function` as it's `custom_conversion_functions's` responsibility to update the shape. Will be pointing to the example for devs understanding
- Added test case with both `custom_conversion_function` and without
- Without `custom_conversion_function`, `custom_shape_function` is mandatory and we raise ValueError while adding custom layer if shape function is not provided
- Will be adding permanent link in example once this PR goes in

**Comment by bhushan23** [comment]
@aseemw all tests are passing locally

**Pull request merged**